tests/react/utils/types.test.tsx

```diff
@@ -0,0 +1,16 @@
+import { it } from '@jest/globals'
+import { useHydrateAtoms } from 'jotai/react/utils'
+import { atom } from 'jotai/vanilla'
+
+it('useHydrateAtoms should not allow invalid atom types when array is passed', () => {
```

Comment:
I thought about adding something for `Map` but I wasn't able to get this type safe because of the limitations Typescript compiler has for `Map`. I'd only be adding a test for `Map` and not `useHydrateAtom`. e.g.
```tsx
it('useHydrateAtoms should not allow Map without generics passed', () => {
  function Component() {
    const countAtom = atom(0)
    const activeAtom = atom(true)

    const map = new Map([
      [countAtom, 'foo'],
      [activeAtom, 0],
    ])
    useHydrateAtoms(map)
  }
  Component
})
```
Really only test that `Map` is not happy:
![Screenshot 2023-05-01 at 9 55 21 PM](https://user-images.githubusercontent.com/525506/235570133-56e5e7d4-4ba4-42aa-9da7-f26290ab5fbe.png)

Unless you can think of something else I should be adding a test for.